### PR TITLE
Turnstile: Add pre-clearance changelog and update csp for it

### PR DESCRIPTION
**Pull request by Le0Developer**
Add changelog for https://blog.cloudflare.com/integrating-turnstile-with-the-cloudflare-waf-to-challenge-fetch-requests

Move CSP into its own reference section to add CSP specific to pre-clearance mode.

### CODE REVIEW

File: content/turnstile/reference/content-security-policy.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Alternatively, add the following values to your CSP header:
+
+- **script-src**: `https://challenges.cloudflare.com`
+- **frame-src**: `https://challenges.cloudflare.com`
+
+We recommend validating your CSP with [Google's CSP Evaluator](https://csp-evaluator.withgoogle.com/).
+
+## With Pre-Clearance
+
+If you are using [Turnstile in pre-clearance mode](https://blog.cloudflare.com/integrating-turnstile-with-the-cloudflare-waf-to-challenge-fetch-requests), Turnstile sets the `cf_clearance` cookie by doing a fetch request to a special endoint in [`/cdn-cgi/`](https://developers.cloudflare.com/fundamentals/reference/cdn-cgi-endpoint/) of your domain.
```
Comment (worenga) [inline content/turnstile/reference/content-security-policy.md]: endoint ==> endpoint

Comment (Le0Developer) [reply content/turnstile/reference/content-security-policy.md]:
```suggestion
If you are using [Turnstile in pre-clearance mode](https://blog.cloudflare.com/integrating-turnstile-with-the-cloudflare-waf-to-challenge-fetch-requests), Turnstile sets the `cf_clearance` cookie by doing a fetch request to a special endpoint in [`/cdn-cgi/`](https://developers.cloudflare.com/fundamentals/reference/cdn-cgi-endpoint/) of your domain.
```
(making it into a suggestion so I can merge directly on github)

File: content/turnstile/frequently-asked-questions.md [hunk 2]
```diff
@@ -13,16 +13,9 @@ structured_data: true
...
 
-We recommend using the nonce-based approach documented with [CSP3](https://w3c.github.io/webappsec-csp/#framework-directive-source-list). Make sure to include your nonce in the `api.js` script tag and we will handle the rest. Cloudflare Turnstile works with **strict-dynamic**.
-
-Alternatively, add the following values to the directives:
-
-* **script-src**: `https://challenges.cloudflare.com`
-* **frame-src**: `https://challenges.cloudflare.com`
-
-We recommend validating your CSP with [Google's CSP Evaluator](https://csp-evaluator.withgoogle.com/).
+For specifics regarding Turnstile, see the [Content Security Policy](/turnstile/reference/content-security-policy/) section.
```
Comment (patriciasantaana) [inline content/turnstile/frequently-asked-questions.md]:
```suggestion
For specifics regarding Turnstile, refer to the [Content Security Policy](/turnstile/reference/content-security-policy/).
```

File: content/turnstile/reference/content-security-policy.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+Alternatively, add the following values to your CSP header:
+
+- **script-src**: `https://challenges.cloudflare.com`
+- **frame-src**: `https://challenges.cloudflare.com`
+
+We recommend validating your CSP with [Google's CSP Evaluator](https://csp-evaluator.withgoogle.com/).
+
+## With Pre-Clearance
+
+If you are using [Turnstile in pre-clearance mode](https://blog.cloudflare.com/integrating-turnstile-with-the-cloudflare-waf-to-challenge-fetch-requests), Turnstile sets the `cf_clearance` cookie by doing a fetch request to a special endpoint in [`/cdn-cgi/`](https://developers.cloudflare.com/fundamentals/reference/cdn-cgi-endpoint/) of your domain.
```
Comment (patriciasantaana) [inline content/turnstile/reference/content-security-policy.md]: including the official docs link (to be published) for pre-clearance instead

```suggestion
## Pre-Clearance support

If you are using [Turnstile in pre-clearance mode](/turnstile/reference/pre-clearance-support/), Turnstile sets the `cf_clearance` cookie by doing a fetch request to a special endpoint in [`/cdn-cgi/`](/fundamentals/reference/cdn-cgi-endpoint/) of your domain.
```

File: data/changelogs/turnstile.yaml [hunk 4]
```diff
@@ -2,6 +2,9 @@
 link: "/turnstile/changelog/"
 productName: Turnstile
 entries:
+- publish_date: '2023-12-18'
+  description: |-
+    - Added [Pre-Clearance mode](https://blog.cloudflare.com/integrating-turnstile-with-the-cloudflare-waf-to-challenge-fetch-requests)
```
Comment (patriciasantaana) [inline data/changelogs/turnstile.yaml]:
```suggestion
    - Added [Pre-Clearance mode](/turnstile/reference/pre-clearance-support/).
```

**Review by patriciasantaana — APPROVED** [review]

**Comment by worenga** [comment]
Thank you @Le0Developer !

**Pull request merged**